[Test] Add smoke test to cover new runtime envar

Kewen12 opened the pull request:
This smoke test is to cover the new runtime envar OMPX_AMD_MEMORY_MANAGER_THRESHOLD_EXP_2 which is designed for users to specify the threshold for the memory size handled by AMDGPUMemoryManager.

Please merge after the internal patch was landed.

Kewen12 committed:
[Test] Add smoke test to cover new runtime envar

This smoke test is to cover the new runtime envar OMPX_AMD_MEMORY_MANAGER_THRESHOLD_EXP_2 which is designed for users to specify the threshold for the memory size handled by AMDGPUMemoryManager.

jplehr commented on test/smoke-dev/rt-memory-threshold-env/AMDGPUMemoryManager_threshold_env.c:
```diff
@@ -0,0 +1,38 @@
+  for (i = 0; i < N; i++)
+    if (a[i] != b[i]) {
+      rc++;
+      printf("Wrong varlue: a[%d]=%d\n", i, a[i]);
```
```suggestion
      printf("Wrong value: a[%d]=%d\n", i, a[i]);
```

Kewen12 replied:
Thanks for catching the typo, JP!

saiislam commented on test/smoke-dev/rt-memory-threshold-env/Makefile:
```diff
@@ -0,0 +1,18 @@
+TESTSRC_AUX  =
+TESTSRC_ALL  = $(TESTSRC_MAIN) $(TESTSRC_AUX)
+RUNENV       += LIBOMPTARGET_DEBUG=1
+RUNENV       += OMPX_AMD_MEMORY_MANAGER_THRESHOLD_EXP_2=20
```
Can you please add a comment specifying the relationship between this value "20" and the expected output value "1048576 B"?

Kewen12 replied:
Thanks for the review. Comment added.

jplehr commented:
@Kewen12 what's the status of this PR?

Kewen12 committed:
Resolved comment

Kewen12 commented:
> @Kewen12 what's the status of this PR?

oops, I missed the notification of the previous comment. Thanks for the reminder. Addressed comment. Please approve if it looks good.

saiislam reviewed (approved):
LGTM.
Thanks @Kewen12 for sticking with it.
